src/transformers/modeling_utils.py

```diff
@@ -2276,11 +2286,25 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                         )
                 dtype_orig = cls._set_default_torch_dtype(torch_dtype)
 
+            # Check if `_keep_in_fp32_modules` is not None
+            # but not force users to have `accelerate`
+            if cls._keep_in_fp32_modules is not None:
+                if not is_accelerate_available() or torch_dtype == torch.bfloat16:
+                    use_keep_in_fp32_modules = False
+                    logger.warning(
+                        " `_keep_in_fp32_modules` is not set to `None` and you don't have `accelerate` installed",
+                        " it is recommended to have `accelerate` installed in this case `pip install accelerate`.",
+                    )
```

Comment:
The warning should only be trigerred when `torch_dtype == torch.float16`.

Reply:
Should be fixed in [8014c34](https://github.com/huggingface/transformers/pull/20683/commits/8014c34e7aa8e1ed3475d4a002e3be46eb3e258e)